lib/core.ps1

```diff
@@ -996,29 +996,15 @@ function show_app($app, $bucket, $version) {
     return $app
 }
 
-function last_scoop_update() {
-    # PowerShell 6 returns an DateTime Object
-    $last_update = (get_config lastupdate)
-
-    if ($null -ne $last_update -and $last_update.GetType() -eq [System.String]) {
-        try {
-            $last_update = [System.DateTime]::Parse($last_update)
-        } catch {
-            $last_update = $null
-        }
-    }
-    return $last_update
-}
-
 function is_scoop_outdated() {
-    $last_update = $(last_scoop_update)
-    $now = [System.DateTime]::Now
-    if($null -eq $last_update) {
-        set_config lastupdate $now.ToString('o')
-        # enforce an update for the first time
+    try{
+        $expireHour = (New-TimeSpan $(get_config lastUpdate)).TotalHours
+        return ($expireHour -ge 3)
+    } catch {
+        # If not System.DateTime
+        set_config lastUpdate ([System.DateTime]::Now.ToString('o')) | Out-Null
```

Comment:
```suggestion
    $now = [System.DateTime]::Now
    try{
        $expireHour = (New-TimeSpan (get_config lastUpdate) $now).TotalHours
        return ($expireHour -ge 3)
    } catch {
        # If not System.DateTime
        set_config lastUpdate ($now.ToString('o')) | Out-Null
```

`New-TimeSpan` implied `[System.DateTime]::Now`, and make it explicit will be better.